src/mender-auth/cli/actions.hpp

```diff
@@ -47,6 +47,10 @@ class Action {
 using ActionPtr = shared_ptr<Action>;
 using ExpectedActionPtr = expected::expected<ActionPtr, error::Error>;
 
+shared_ptr<MenderKeyStore> KeystoreFromConfig(
+	const conf::MenderConfig &config, const string &passphrase);
```

Comment:
This helper we should rather move to `mender::client_shared::conf` so that it can be used from `mender-update` without requiring this coupling. This way we are also free to revert 39ae079890f109c7f06234576e1d9dbc2d74f12 which I don't think it has value for production, only for prototypoing.

Reply:
I thought about that, but it doesn't work without moving more code from mender-auth. The problem is that this function depends on the `MenderKeyStore` type, and you can't include that file from `mender::client_shared::conf` because it would be a circular dependency. Then we would also need to move that type and the associated functions also from mender-auth to client_shared, and I figured that all the related code was too specific to mender-auth to move into the shared section. I could be convinced otherwise I suppose, but that was my reasoning.